plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -332,6 +348,31 @@ public static ReaderPageSource createPageSource(
         return Optional.of(new GroupType(baseType.getRepetition(), baseType.getName(), ImmutableList.of(type)));
     }
 
+    private static ColumnIndexStore getColumnIndexStore(Predicate parquetPredicate, ParquetDataSource dataSource, BlockMetaData blockMetadata, Map<List<String>, RichColumnDescriptor> descriptorsByPath, boolean readColumnIndexFilter)
+    {
+        if (!readColumnIndexFilter || parquetPredicate == null || !(parquetPredicate instanceof TupleDomainParquetPredicate)) {
+            return null;
+        }
+
+        boolean hasColumnIndex = false;
+        for (ColumnChunkMetaData column : blockMetadata.getColumns()) {
+            if (column.getColumnIndexReference() != null && column.getOffsetIndexReference() != null) {
+                hasColumnIndex = true;
+                break;
+            }
+        }
+
+        if (!hasColumnIndex) {
+            return null;
+        }
+
+        Set<ColumnPath> paths = new HashSet<>();
+        for (List<String> path : descriptorsByPath.keySet()) {
+            paths.add(ColumnPath.get(path.toArray(new String[0])));
+        }
+        return ColumnIndexStoreImpl.create(dataSource, blockMetadata, paths);
```

Comment:
@martint's comment here:
> ```java
>         for (ColumnChunkMetaData column : blockMetadata.getColumns()) {
>             if (column.getColumnIndexReference() != null && column.getOffsetIndexReference() != null) {
>                 Set<ColumnPath> paths = descriptorsByPath.keySet().stream()
>                         .map(path -> ColumnPath.get(path.toArray(new String[0])))
>                         .collect(toImmutableSet());
>                 return ColumnIndexStoreImpl.create(dataSource, blockMetadata, paths);
>             }
>         }
>         return null;
> ```

Reply:
Or you could go all-in with streams, especially if the method returns an `Optional`:
```java
        return blockMetadata.getColumns().stream()
                .filter(column -> column.getColumnIndexReference() != null && column.getOffsetIndexReference() != null)
                .findAny()
                .map(ignored -> ColumnIndexStoreImpl.create(
                        dataSource,
                        blockMetadata,
                        descriptorsByPath.keySet().stream()
                                .map(path -> ColumnPath.get(path.toArray(String[]::new)))
                                .collect(toImmutableSet())));
```